mne/coreg.py

```diff
@@ -1242,3 +1245,613 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+               d['kind'] == FIFF.FIFFV_POINT_HPI and
+               d['coord_frame'] == FIFF.FIFFV_COORD_HEAD]
+        out = np.empty((0, 3)) if len(out) == 0 else np.array(out)
+        return out
+
+    @property
+    def eeg_points(self):
+        if not self._info or not self._info['dig']:
+            return np.empty((0, 3))
+
+        out = [d['r'] for d in self._info['dig'] if
+               d['kind'] == FIFF.FIFFV_POINT_EEG and
+               d['coord_frame'] == FIFF.FIFFV_COORD_HEAD]
+        out = np.empty((0, 3)) if len(out) == 0 else np.array(out)
+        return out
+
+    def _cardinal_point(self, ident):
+        """Coordinates for a cardinal point."""
+        if not self._info or not self._info['dig']:
+            return np.zeros((1, 3))
+
+        for d in self._info['dig']:
+            if d['kind'] == FIFF.FIFFV_POINT_CARDINAL and d['ident'] == ident:
+                return d['r'][None, :]
+        return np.zeros((1, 3))
+
+    @property
+    def nasion(self):
+        return self._cardinal_point(FIFF.FIFFV_POINT_NASION)
+
+    @property
+    def lpa(self):
+        return self._cardinal_point(FIFF.FIFFV_POINT_LPA)
+
+    @property
+    def rpa(self):
+        return self._cardinal_point(FIFF.FIFFV_POINT_RPA)
+
+
+class _Surf(object):
+    def __init__(self, rr=None, nn=None, tris=None):
+        self.rr = np.empty((0, 3)) if rr is None else rr
+        self.nn = np.empty((0, 3)) if nn is None else nn
+        self.tris = np.empty((0, 3)) if tris is None else tris
+
+
+class _SurfaceSource(object):
+    def __init__(self, filename=None):
+        if filename is not None and op.exists(filename):
+            if filename.endswith('.fif'):
+                bem = read_bem_surfaces(filename, verbose=False)[0]
+            else:
+                try:
+                    bem = read_surface(filename, return_dict=True)[2]
+                    bem['rr'] *= 1e-3
+                    complete_surface_info(bem, copy=False)
+                except Exception:
+                    raise ValueError(
+                        "Error loading surface from %s (see "
+                        "Terminal for details)." % filename)
+            self.surf = _Surf(rr=bem['rr'], tris=bem['tris'], nn=bem['nn'])
+        else:
+            self.surf = _Surf()
+
+
+class _FiducialsSource(object):
+    def __init__(self, filename=None):
+        self.filename = filename
+        self.mni_points = None
+
+    @property
+    def points(self):
+        if self.filename is None or not op.exists(self.filename):
+            return self.mni_points
+        try:
+            return _fiducial_coords(*read_fiducials(self.filename))
+        except Exception as err:
+            raise ValueError(
+                "Error reading fiducials from %s: %s (See terminal "
+                "for more information)" % (self.filename, str(err)),
+                "Error Reading Fiducials")
+
+
+class _MRIHeadWithFiducialsModel(object):
```

Comment:
Don't need this, just load head surfaces directly in the Coreg class. Maybe some of this should become a private function, but really we should refactor with the MRI-head-surface-finding code in `mne.viz.plot_alignment`